z-client-mu-plugins.php

```diff
@@ -1,107 +1,85 @@
 <?php
 
-// Note: This file is prefixed with `z-` for load order
-
-function wpcom_vip_load_client_mu_plugins() {
-	static $loaded = false;
+/**
+ * Plugin Name: VIP Client mu-plugins
+ * Description: Helper plugin to load custom mu-plugins in the `client-mu-plugins` dir.
+ * Author: Automattic
+ */
 
-	// Prevent running this multiple times
-	if ( $loaded ) {
-		return;
-	}
-
-	$loaded = true;
+// Note: This file is prefixed with `z-` for load order
 
-	// Code below is adapted from wp_get_mu_plugins()
+/**
+ * Gets PHP files in the client-mu-plugins folder.
+ *
+ * The code for this function is adapted from `wp_get_mu_plugins()`
+ */
+function wpcom_vip_get_client_mu_plugins() {
 	$client_mu_plugins = [];
 
 	if ( ! is_dir( WPCOM_VIP_CLIENT_MU_PLUGIN_DIR ) ) {
 		return;
 	}
 
-	$dh = opendir( WPCOM_VIP_CLIENT_MU_PLUGIN_DIR );
-	if ( ! $dh ) {
+	$dir_handle = opendir( WPCOM_VIP_CLIENT_MU_PLUGIN_DIR );
+	if ( ! $dir_handle ) {
 		return;
 	}
 
 	do {
-		$plugin = readdir( $dh );
+		$plugin = readdir( $dir_handle );
 		if ( false === $plugin ) {
 			break;
 		}
 
-		if ( substr( $plugin, -4 ) === '.php' ) {
+		$is_php_file = substr( $plugin, -4 ) === '.php';
+		if ( $is_php_file ) {
 			$client_mu_plugins[] = WPCOM_VIP_CLIENT_MU_PLUGIN_DIR . '/' . $plugin;
 		}
 	} while ( false !== $plugin );
 
-	closedir( $dh );
+	closedir( $dir_handle );
 
 	// Make sure plugins load in a consistent, predictable order
 	sort( $client_mu_plugins );
 
-	foreach ( $client_mu_plugins as $plugin ) {
-		include_once( $plugin );
-	}
-}
-
-// Load the plugins
-// TODO: move out of function scope to avoid issues with globals not being properly set
-wpcom_vip_load_client_mu_plugins();
-
-function wpcom_vip_get_client_mu_plugins() {
-	$wp_plugins = array();
-	$plugin_files = array();
-
-	if ( ! is_dir( WPCOM_VIP_CLIENT_MU_PLUGIN_DIR ) ) {
-		return $wp_plugins;
-	}
-
-	if ( $plugins_dir = @opendir( WPCOM_VIP_CLIENT_MU_PLUGIN_DIR ) ) {
-		while ( ( $file = readdir( $plugins_dir ) ) !== false ) {
-			if ( substr( $file, -4 ) === '.php' ) {
-				$plugin_files[] = $file;
-			}
-		}
-	} else {
-		return $wp_plugins;
-	}
-
-	@closedir( $plugins_dir );
-
-	if ( empty( $plugin_files ) ) {
-		return $wp_plugins;
-	}
-
-	return $plugin_files;
+	return $client_mu_plugins;
 }
 
 function wpcom_vip_get_client_mu_plugins_data() {
-	$plugin_files = wpcom_vip_get_client_mu_plugins();
+	$client_mu_plugins_files = wpcom_vip_get_client_mu_plugins();
 
-	if ( empty( $plugin_files ) ) {
-		return $plugin_files;
+	if ( empty( $client_mu_plugins_files ) ) {
+		return [];
 	}
 
-	foreach ( $plugin_files as $plugin_file ) {
-		if ( ! is_readable( WPCOM_VIP_CLIENT_MU_PLUGIN_DIR . "/$plugin_file" ) ) {
+	$client_mu_plugins_data = [];
+
+	foreach ( $client_mu_plugins_files as $plugin_file ) {
+		if ( ! is_readable( WPCOM_VIP_CLIENT_MU_PLUGIN_DIR . '/' . $plugin_file ) ) {
```

Comment:
For some reason this path is a little strange for me when running this locally: 

`/home/scott/Sites/wordpress/wp-content/client-mu-plugins//home/scott/Sites/wordpress/wp-content/client-mu-plugins/test-plugin.php`

Not sure what has really changed (could be something on my local setup) but it is causing the endpoint to fail for me.